Objects/unicodeobject.c

```diff
@@ -1429,8 +1429,8 @@ _copy_characters(PyObject *to, Py_ssize_t to_start,
     assert(PyUnicode_Check(from));
     assert(from_start + how_many <= PyUnicode_GET_LENGTH(from));
 
-    assert(PyUnicode_Check(to));
-    assert(to_start + how_many <= PyUnicode_GET_LENGTH(to));
+    assert(to == NULL || PyUnicode_Check(to));
+    assert(how_many == 0 || (to_start + how_many <= PyUnicode_GET_LENGTH(to)));
```

Comment:
Actually, I've had a better idea:

```C
assert(to == NULL || PyUnicode_Check(to));
if (how_many == 0) {
    return 0;
}

assert(to != NULL);
assert(to_start + how_many <= PyUnicode_GET_LENGTH(to));
```

That way, we also guarantee that we do not have a `to == NULL` in the `else` branch.

Reply:
Maybe so:
```c
if (how_many == 0) {
    return 0;
}

assert(PyUnicode_Check(to));
assert(to_start + how_many <= PyUnicode_GET_LENGTH(to));
```
`assert(to == NULL || PyUnicode_Check(to))` - it doesn't make sense.
* if `to == NULL`, then `PyUnicode_Check(to)` will not be performed.
* if `to != NULL`, then `PyUnicode_Check(to)` will already check `to != NULL`.